fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/visitor/UnionVisitor.java

```diff
@@ -93,16 +95,30 @@ public RecordQueryPlan postVisit(@Nonnull final RecordQueryPlan recordQueryPlan)
                     }
                     oldPlan = ((RecordQueryFilterPlan) oldPlan).getChild();
                 }
+                @Nullable RecordQueryFetchFromPartialRecordPlan.FetchIndexRecords currentFetchIndexRecords = resolveFetchIndexRecordsFromPlan(oldPlan);
```

Comment:
It is a little awkward that this logic needs to be repeated in the `IntersectionVisitor` and the `UnionVisitor`. I'm not sure if there's a better place, though, to put a helper method that extracts the common fetch type (or returns `null`).

Reply:
`resolveFetchIndexRecordsFromPlan()` is in the base class. `Union` and `Intersection` are slightly differently but still need to walk the legs and do some (different) logs.